torch/autograd/profiler_util.py

```diff
@@ -127,11 +127,18 @@ def _populate_cpu_children(self):
                         # this can't be a parent
                         current_events.pop()
                     else:
-                        parent.append_cpu_child(event)
-                        assert (
-                            event.cpu_parent is None
-                        ), f"There is already a CPU parent event for {event.key}"
-                        event.set_cpu_parent(parent)
+                        is_contain = False
+                        for child in parent.cpu_children:
```

Comment:
We need to be careful here, this increases the time complexity and may cause significant post processing time. Can you also update the algorithm comments in lines 101 - 112.